src/test/java/com/easypost/AddressTest.java

```diff
@@ -0,0 +1,152 @@
+        
+        verificationList.add("true");
+        addressData.put("verify_strict", verificationList);
+
+        Address address = Address.create(addressData);
+
+        assertTrue(address instanceof Address);
+        assertTrue(address.getId().startsWith("adr_"));
+        assertEquals("388 TOWNSEND ST APT 20", address.getStreet1());
+    }
+
+    /**
+     * Test retrieving an address.
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    @Test
+    public void testRetrieve() throws EasyPostException{
+        Address retrievedAddress = Address.retrieve(address.getId());
+
+        assertTrue(retrievedAddress instanceof Address);
+        assertTrue(retrievedAddress.getId().startsWith("adr_"));
+        assertEquals(address.getId(), retrievedAddress.getId());
+    }
+
+    /**
+     * Test retrieving all addresses.
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    @Test
+    public void testAll() throws EasyPostException{
+        Map<String, Object> params = new HashMap<>();
+
+        params.put("page_size", Fixture.pageSize());
+
+        AddressCollection addresses = Address.all(params);
+
+        List<Batch> addressesArray = addresses.getAddresses();
+
+        assertNotNull(addresses.getHasMore());
+        assertTrue(addressesArray.size() <= 5);
+        for(Batch address: addressesArray) {
+            assertTrue(address.getId().startsWith("adr_"));
+        }
+    }
+
+    /**
+     * Test creating a verified address.
+     * We purposefully pass in slightly incorrect data to get the corrected address back once verified.
+     *
+     *  @throws EasyPostException when the request fails.
+     */
+    @Test
+    public void testCreateVerify() throws EasyPostException {
+        Address address = Address.create(Fixture.incorrectAddressToVerify());
+
+        assertTrue(address instanceof Address);
+        assertTrue(address.getId().startsWith("adr_"));
+        assertEquals("417 MONTGOMERY ST STE 500", address.getStreet1());
+    }
+
+    /**
+     * Test creating a verified address.
+     * We purposefully pass in slightly incorrect data to get the corrected address back once verified.
+     *
+     *  @throws EasyPostException when the request fails.
+     */
+    @Test
+    public void testCreateAndVerify() throws EasyPostException {
+        Address address = Address.createAndVerify(Fixture.incorrectAddressToVerify());
+
+        assertTrue(address instanceof Address);
+        assertTrue(address.getId().startsWith("adr_"));
+        assertEquals("417 MONTGOMERY ST STE 500", address.getStreet1());
+        assertEquals("CA", address.getState());
+        assertEquals("SAN FRANCISCO", address.getCity());
+        assertEquals("94104-1100", address.getZip());
+        assertEquals("4151234567", address.getPhone());
+        assertEquals("EASYPOST", address.getCompany());
+        assertEquals("US", address.getCountry());
```

Comment:
Please follow the PHP library - all these additional assertions are unnecessary and complicate the test. `street1` is sufficient here as it corrects, thus we can infer that the rest of the address was also corrected.

After chatting, this was done to increase coverage (thank you!); however, for simplicity, let's stick to the example and remove all the additional assertions for now - we may just configure the coverage platform to ignore getters/setters.